src/webdrivers/browser_factory.py

```diff
@@ -0,0 +1,55 @@
+from typing import Union
+
+from selenium import webdriver
+from loguru import logger
+
+from config import BROWSER_TYPE_CONFIG
+from src.webdrivers.base_browser import BrowserType
+from src.webdrivers.chrome import Chrome
+from src.webdrivers.firefox import Firefox
+
+
+class BrowserFactory:
+    """Factory class for creating browser instances"""
+    _browser_type: BrowserType = BrowserType.CHROME  # Default browser type
+
+    _browsers = {
```

Comment:
removed, thanks for pointing this out.

Reply:
not removed yet

Reply:
sample code
```py
class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

# Checking if a specific member is in the enum
print(Color.RED in Color)  # Output: True

```

Reply:
We need to have a mapping in here to initialize the browser instance with the class name. Otherwise, it will result circular import issues from the base_browser file. Please let me know what you think.

Reply:
oh that means there is a design problem in this. let me have a look at the code.

Reply:
base_browser is not using browser_factory. I don't see any circular issue. can you add stacktrace here.

Reply:
even in current code you are importing browser type. line 7

Reply:
you can update it to in.

Reply:
Sorry for the confussion.  I meant if we want to have a single source of truth, we can change the class `BrowserType` in `base_browser.py` from 
``` python
class BrowserType(Enum):
    """Enum for supported browser types"""
    CHROME = "chrome"
    FIREFOX = "firefox"
```

to 

``` python
from src.webdrivers.chrome import Chrome
from src.webdrivers.firefox import Firefox

class BrowserType(Enum):
    """Enum for supported browser types"""
    CHROME = Chrome
    FIREFOX = Firefox
```

then in `BrowserFactory.get_browser()` we dont need to do additional name comparison and can just initialize the browser instance direcrtly.

But the newly added 
```
from src.webdrivers.chrome import Chrome
from src.webdrivers.firefox import Firefox
```
in `base_browser.py` will result a circular import issues, as `Chrome.py` import `base_browser.py` and `base_browser.py` import `Chrome.py`.

Not sure if there is a better solution for this.

Reply:
@Tverous I will take care it.

Reply:
@Tverous made changes. can you test your branch once?

Reply:
dc875e67b0a1f74377cd3e4a463a32c8c5739719 breaks the functionality.

Reply:
Refactored the browser type by changing certain argument types to `str` while maintaining readability. All issues should have been resolved, consolidating everything into a single source of truth in `browser_type.py`.

Reply:
why is this change required? exppain why you moved from ENUM to string?

Reply:
To prevent circular imports, the impact is minimal since we will convert those values to strings within the class. Those arguments will be used solely for creating the profile directory and for logging purposes.

Reply:
looks like these are handled @Tverous good work